### PR TITLE
[tests-only][full-ci]move login step after all API query is done

### PR DESCRIPTION
**Pull request by S-Panta**
## Description

As per https://github.com/owncloud/web/pull/11456#pullrequestreview-2289864737 , the login step will be done only after all API of the `Given` statement.  

For the scenario involving creating the link, `roleDisplayText` and `securePassword` are removed from the Link object because these are just used for storing link-related information, and to access them, one has to create a new object. 

## Related Issue

- Fixes https://github.com/owncloud/web/issues/11496

## How Has This Been Tested?

- test environment:
- locally and CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: tests/e2e/cucumber/features/file-action/rename.feature [hunk 1]
```diff
@@ -21,7 +21,7 @@ Feature: rename
     And "Alice" creates a public link of following resource using API
       | resource | role     | password |
       | folder   | Can edit | %public% |
-    And "Alice" opens the "files" app
```
Comment (ScharfViktor) [inline tests/e2e/cucumber/features/file-action/rename.feature:24]: could you move it to line 42 to get 
```
And "Alice" logs in
And "Alice" navigates to the shared with others page
And "Alice" opens folder "folder"
```

**Review by ScharfViktor — COMMENTED** [review]
other is fine 👍

File: tests/e2e/cucumber/steps/ui/public.ts [hunk 2]
```diff
@@ -31,11 +32,10 @@ When(
   async function (this: World, stepUser: string, password: string): Promise<void> {
     const { page } = this.actorsEnvironment.getActor({ key: stepUser })
     const pageObject = new objects.applicationFiles.page.Public({ page })
-    const linkObject = new objects.applicationFiles.Link({ page })
     if (password === '%copied_password%') {
       password = await page.evaluate('navigator.clipboard.readText()')
     } else {
-      password = password === '%public%' ? linkObject.securePassword : password
+      password = '%public%' === password ? securePassword : password
```
Comment (saw-jan) [inline tests/e2e/cucumber/steps/ui/public.ts:38]: this is more preferable
```suggestion
      password = password === '%public%' ? securePassword : password
```

**Review by saw-jan — CHANGES_REQUESTED** [review]

File: tests/e2e/cucumber/features/file-action/rename.feature [hunk 3]
```diff
@@ -8,7 +8,7 @@ Feature: rename
       | id    |
       | Alice |
       | Brian |
-    And "Alice" logs in
+
```
Comment (amrita-shrestha) [inline tests/e2e/cucumber/features/file-action/rename.feature:11]:
```suggestion
```

**Review by saw-jan — APPROVED** [review]
LGTM

**Review by ScharfViktor — APPROVED** [review]
👍

**Pull request merged**